plugins/rebalance/transaction.go

```diff
@@ -0,0 +1,52 @@
+package rebalance
+
+import (
+	"github.com/gluster/glusterd2/glusterd2/daemon"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	rebalanceapi "github.com/gluster/glusterd2/plugins/rebalance/api"
+)
+
+type actionType uint16
+
+const (
+	actionStart actionType = iota
+	actionStop
+)
+
+func rebalanceCommand(c transaction.TxnCtx, cmd actionType) error {
+	var rinfo rebalanceapi.RebalInfo
+	err := c.Get("rinfo", &rinfo)
+	if err != nil {
+		return err
+	}
+
+	rebalanceProcess, err := NewRebalanceProcess(rinfo)
+	if err != nil {
+		return err
+	}
+
+	switch cmd {
+	case actionStart:
+		err = daemon.Start(rebalanceProcess, true)
+
+		// Create rpc client connection
+		_, err := daemon.GetRPCClient(rebalanceProcess)
+		if err != nil {
+			c.Logger().WithError(err).WithField(
+				"volume", rinfo.Volname).Error("rpc creation failed")
+			return err
+		}
+
+	case actionStop:
+		err = daemon.Stop(rebalanceProcess, true)
```

Comment:
This needs confirmation from DHT team. Stopping Daemon may be problem here. Ongoing file copy may get affected. My guess here is you have to call `exec.Command` with same arguments as Rebalance start (rebal.Cmd will be different in this case). 

@nbalacha need your help in understanding Stop functionality.

Reply:
If we are sending the stop command to the rebalance process, it takes care of finishing any ongoing file migrations before terminating itself.

What does daemon.Stop do?

Reply:
`daemon.Stop` sends SIGTERM to the daemon.

Reply:
Then this is incorrect.

Reply:
@nbalacha Can you briefly tell us about how glusterd2 should implement signalling rebalance to stop ? Is there a RPC call that must be sent ?

Reply:
Will do. I will send the info tomorrow .